### DISCUSSION TITLE
HasManyRelationManager: How to update ownerRecord when delete?

### DISCUSSION THREAD

**Discussion by diazsasak** [filamentphp/filament#1461] [General]
Hi, Thank you for this great package, it make developer happy and do all repetitive work.

I have ProductResource that have HasMany relation to the stock history, when i add stock history i can update the ownerRecord stock using mutateFormDataBeforeCreate. But when i delete a stock history i want to revert stock amount on ownerRecord. Is there any way / lifecycle hook for delete? 

Thank you

**Comment by diazsasak** [comment]
ok i think laravel has the answer :) https://laravel.com/docs/8.x/eloquent#events

**Comment by diazsasak** [comment]
can we tell the parent (EditRecord) component to refresh?

**Reply by danharrin** [reply]
You could emit an event and register a `$refres` listener for it on the Edit page, maybe?
